README.md
# Task::Date::Holidays

[![CPAN version](https://badge.fury.io/pl/Task-Date-Holidays.svg)](http://badge.fury.io/pl/Task-Date-Holidays)
![Build Status](https://github.com/jonasbn/perl-task-date-holidays/actions/workflows/ci.yml/badge.svg)
![Spellcheck](https://github.com/jonasbn/perl-task-date-holidays/actions/workflows/spellcheck.yml/badge.svg)
![Markdownlint check](https://github.com/jonasbn/perl-task-date-holidays/actions/workflows/markdownlint.yml/badge.svg)

# NAME

Task::Date::Holidays - Work distribution for holiday modules in the Date::Holidays namespace

# SYNOPSIS

    $ tar xvzf Task-Date-Holidays-0.12.tgz

    $ cd Task-Date-Holidays-0.12

    $ perl Build.PL

    $ ./Build install

    # hack, hack, hack

    # take some holiday

# DESCRIPTION

This distribution is intended to provide an easy way to install all of the Date::Holidays::\* distributions.

Bundling these makes it easier for me, in the work related to maintaining [Date::Holidays](https://metacpan.org/pod/Date%3A%3AHolidays).

If your distribution is missing send me a pull request or just let me know.

If you want to provide a distribution to the namespace, please see [Date::Holidays](https://metacpan.org/pod/Date%3A%3AHolidays) for more information.

Do note that because a distribution is listed in the below list, does not necessarily make it compatible with the Date::Holidays API. Distributions listed here indicate that a given distribution is either compatible or being considered for inclusion.

Please see [Date::Holidays](https://metacpan.org/pod/Date%3A%3AHolidays)

# CONTENTS

- [Date::Holidays::AT](https://metacpan.org/pod/Date%3A%3AHolidays%3A%3AAT)
- [Date::Holidays::AU](https://metacpan.org/pod/Date%3A%3AHolidays%3A%3AAU)
- [Date::Holidays::AW](https://metacpan.org/pod/Date%3A%3AHolidays%3A%3AAW)
- [Date::Holidays::BR](https://metacpan.org/pod/Date%3A%3AHolidays%3A%3ABR)
- [Date::Holidays::BY](https://metacpan.org/pod/Date%3A%3AHolidays%3A%3ABY)
- [Date::Holidays::CA](https://metacpan.org/pod/Date%3A%3AHolidays%3A%3ACA)
- [Date::Holidays::CA\_ES](https://metacpan.org/pod/Date%3A%3AHolidays%3A%3ACA_ES)
- [Date::Holidays::CN](https://metacpan.org/pod/Date%3A%3AHolidays%3A%3ACN)
- [Date::Holidays::CZ](https://metacpan.org/pod/Date%3A%3AHolidays%3A%3ACZ)
- [Date::Holidays::DE](https://metacpan.org/pod/Date%3A%3AHolidays%3A%3ADE)
- [Date::Holidays::DK](https://metacpan.org/pod/Date%3A%3AHolidays%3A%3ADK)
- [Date::Holidays::ES](https://metacpan.org/pod/Date%3A%3AHolidays%3A%3AES)
- [Date::Holidays::FR](https://metacpan.org/pod/Date%3A%3AHolidays%3A%3AFR)
- [Date::Holidays::GB](https://metacpan.org/pod/Date%3A%3AHolidays%3A%3AGB)
- [Date::Holidays::KR](https://metacpan.org/pod/Date%3A%3AHolidays%3A%3AKR)
- [Date::Holidays::KZ](https://metacpan.org/pod/Date%3A%3AHolidays%3A%3AKZ)
- [Date::Holidays::NL](https://metacpan.org/pod/Date%3A%3AHolidays%3A%3ANL)
- [Date::Holidays::NO](https://metacpan.org/pod/Date%3A%3AHolidays%3A%3ANO)
- [Date::Holidays::NZ](https://metacpan.org/pod/Date%3A%3AHolidays%3A%3ANZ)
- [Date::Holidays::PL](https://metacpan.org/pod/Date%3A%3AHolidays%3A%3APL)
- [Date::Holidays::PT](https://metacpan.org/pod/Date%3A%3AHolidays%3A%3APT)
- [Date::Holidays::RU](https://metacpan.org/pod/Date%3A%3AHolidays%3A%3ARU)
- [Date::Holidays::SK](https://metacpan.org/pod/Date%3A%3AHolidays%3A%3ASK)
- [Date::Holidays::UA](https://metacpan.org/pod/Date%3A%3AHolidays%3A%3AUA)
- [Date::Holidays::UK](https://metacpan.org/pod/Date%3A%3AHolidays%3A%3AUK)
- [Date::Holidays::US](https://metacpan.org/pod/Date%3A%3AHolidays%3A%3AUS)
- [Date::Holidays::USFederal](https://metacpan.org/pod/Date%3A%3AHolidays%3A%3AUSFederal)
- [Date::Japanese::Holiday](https://metacpan.org/pod/Date%3A%3AJapanese%3A%3AHoliday)

# SEE ALSO

- [Date::Holidays](https://metacpan.org/pod/Date%3A%3AHolidays)
- [Date::Holidays::Adapter](https://metacpan.org/pod/Date%3A%3AHolidays%3A%3AAdapter)
- [Date::Holidays::Abstract](https://metacpan.org/pod/Date%3A%3AHolidays%3A%3AAbstract)
- [Date::Holidays::Super](https://metacpan.org/pod/Date%3A%3AHolidays%3A%3ASuper)

# CONFIGURATION AND ENVIRONMENT

No special configuration or environment is required.

# ISSUE REPORTING

Please report any bugs or feature requests using GitHub

- [Github Issues](https://github.com/jonasbn/perl-task-date-holidays/issues)

# AUTHOR

Jonas B., (jonasbn) - `<jonasbn@cpan.org>`

# ACKNOWLEDGEMENTS

Contributors and others in alphabetical order

- Alberto Simoes (AMBS), for patching Date::Holidays::PT which was broken
- Alexander Nalobin (NALOBIN), patch for inclusion of Date::Holidays::RU
- Gabor Szabo (SZABGAB), patch assisting META data generation
- Slaven Rezic (SREZIC), reporting broken tests
- All contributors to the Date::Holidays::\* namespace

# LICENSE AND COPYRIGHT

Task-Date-Holidays and related modules are (C) by Jonas B., (jonasbn)
2014-2022

Task-Date-Holidays and related modules are released under the Artistic License 2.0


Changes
Changes file for Task-Date-Holidays

HEAD

- Added Date::Holidays::US for evaluation

0.12 2022-07-08 Bug fix release, update not required

- Corrected tests, which had not been updated accordingly

0.11 2022-03-08 Feature release, update not required

- Specified requirement for Date::Holidays::FR to be version 0.04

- Extended the test requirements to include Date::Holidays

0.10 2020-09-29 Maintenance release, update not required

- Added missing test requirements to the configuration
  Test::Fatal and Test::Class

0.09 2020-09-27 Feature release, update not required

- Added Date::Holidays::AW, Aruba

- Added Date::Holidays::KZ, Kazakhstan

- Added Date::Holidays::NL, Netherlands

- Reenabled Date::Holidays::CN, China

- Added test suite


0.08 2020-04-22 Feature release, update not required

- Added Date::Holidays::UA, Ukraine


0.07 2017-06-05 Maintenance release, update not required

- Added CONTRIBUTING.md

- Updated Dist::Zilla config to contemporary version

- Added Gitter link (Date::Holidays channel)

- Added MANIFEST.SKIP from my Gist: https://gist.github.com/jonasbn/aa6646544f2233de39f34c49532215ad

- Changed bug reporting to Github issues


0.06 2017-06-03 Feature release, update not required

- Added Date::Holidays::BY, Belarus

- Added Date::Japanese::Holiday, Japan - had been left out?

- Added Date::Holidays::UK, United Kingdom - had also been left out?


0.05 2015-08-22 Feature release, update not required

- Re-enabled Date::Holidays::PT, thanks to AMBS for patching it
  so it will work again, thanks to AMBS for patch resulting in release 0.03 (see RT:98441)

  https://rt.cpan.org/Ticket/Display.html?id=98441

- Date::Holidays::CZ disabled (broken build)

- Date::Holidays::CN disabled (broken build)


0.04 2015-01-31 Feature release, update not required

- Adding Date::Holidays::RU, contribution by Alexander Nalobin


0.03 2014-08-30 Maintenance release, update not required

- Merged changes from Gabor Szabo adding handling of META data on
  Github and generation of META.json


0.02 2014.08.29 Maintenance release, update not required

- Fixed POD error (missing heading)

- Added Perl version requirement, POD syntax use requires 5.6


0.01 2014.08.28 Initial release

- First shot at a work package for Date::Holidays

- Lists:

    Date::Holidays::AT = 0
    Date::Holidays::NO = 0
    Date::Holidays::DK = 0
    Date::Holidays::DE = 0
    Date::Holidays::GB = 0
    Date::Holidays::PT = 0
    Date::Holidays::ES = 0
    Date::Holidays::PL = 0
    Date::Holidays::CZ = 0
    Date::Holidays::KR = 0
    Date::Holidays::SK = 0
    Date::Holidays::FR = 0
    Date::Holidays::BR = 0
    Date::Holidays::CA_ES = 0
    Date::Holidays::USFederal = 0
    Date::Holidays::CA = 0
    Date::Holidays::CN = 0
    Date::Holidays::NZ = 0
    Date::Holidays::AU = 0


lib/Task/Date/Holidays.pm
package Task::Date::Holidays;

use strict;
use warnings;
use vars qw($VERSION);
use 5.006;

$VERSION = '0.12';

1;

__END__

=pod

=encoding UTF-8

=begin markdown

# Task::Date::Holidays

[![CPAN version](https://badge.fury.io/pl/Task-Date-Holidays.svg)](http://badge.fury.io/pl/Task-Date-Holidays)
![Build Status](https://github.com/jonasbn/perl-task-date-holidays/actions/workflows/ci.yml/badge.svg)
![Spellcheck](https://github.com/jonasbn/perl-task-date-holidays/actions/workflows/spellcheck.yml/badge.svg)
![Markdownlint check](https://github.com/jonasbn/perl-task-date-holidays/actions/workflows/markdownlint.yml/badge.svg)

=end markdown

=head1 NAME

Task::Date::Holidays - Work distribution for holiday modules in the Date::Holidays namespace

=head1 SYNOPSIS

    $ tar xvzf Task-Date-Holidays-0.12.tgz

    $ cd Task-Date-Holidays-0.12

    $ perl Build.PL

    $ ./Build install

    # hack, hack, hack

    # take some holiday

=head1 DESCRIPTION

This distribution is intended to provide an easy way to install all of the Date::Holidays::* distributions.

Bundling these makes it easier for me, in the work related to maintaining L<Date::Holidays>.

If your distribution is missing send me a pull request or just let me know.

If you want to provide a distribution to the namespace, please see L<Date::Holidays> for more information.

Do note that because a distribution is listed in the below list, does not necessarily make it compatible with the Date::Holidays API. Distributions listed here indicate that a given distribution is either compatible or being considered for inclusion.

Please see L<Date::Holidays>

=head1 CONTENTS

=over

=item * L<Date::Holidays::AT>

=item * L<Date::Holidays::AU>

=item * L<Date::Holidays::AW>

=item * L<Date::Holidays::BR>

=item * L<Date::Holidays::BY>

=item * L<Date::Holidays::CA>

=item * L<Date::Holidays::CA_ES>

=item * L<Date::Holidays::CN>

=item * L<Date::Holidays::CZ>

=item * L<Date::Holidays::DE>

=item * L<Date::Holidays::DK>

=item * L<Date::Holidays::ES>

=item * L<Date::Holidays::FR>

=item * L<Date::Holidays::GB>

=item * L<Date::Holidays::KR>

=item * L<Date::Holidays::KZ>

=item * L<Date::Holidays::NL>

=item * L<Date::Holidays::NO>

=item * L<Date::Holidays::NZ>

=item * L<Date::Holidays::PL>

=item * L<Date::Holidays::PT>

=item * L<Date::Holidays::RU>

=item * L<Date::Holidays::SK>

=item * L<Date::Holidays::UA>

=item * L<Date::Holidays::UK>

=item * L<Date::Holidays::US>

=item * L<Date::Holidays::USFederal>

=item * L<Date::Japanese::Holiday>

=back

=head1 SEE ALSO

=over

=item * L<Date::Holidays>

=item * L<Date::Holidays::Adapter>

=item * L<Date::Holidays::Abstract>

=item * L<Date::Holidays::Super>

=back

=head1 CONFIGURATION AND ENVIRONMENT

No special configuration or environment is required.

=head1 ISSUE REPORTING

Please report any bugs or feature requests using GitHub

=over

=item * L<Github Issues|https://github.com/jonasbn/perl-task-date-holidays/issues>

=back

=head1 AUTHOR

Jonas B., (jonasbn) - C<< <jonasbn@cpan.org> >>

=head1 ACKNOWLEDGEMENTS

Contributors and others in alphabetical order

=over

=item * Alberto Simoes (AMBS), for patching Date::Holidays::PT which was broken

=item * Alexander Nalobin (NALOBIN), patch for inclusion of Date::Holidays::RU

=item * Gabor Szabo (SZABGAB), patch assisting META data generation

=item * Slaven Rezic (SREZIC), reporting broken tests

=item * All contributors to the Date::Holidays::* namespace

=back

=head1 LICENSE AND COPYRIGHT

Task-Date-Holidays and related modules are (C) by Jonas B., (jonasbn)
2014-2022

Task-Date-Holidays and related modules are released under the Artistic License 2.0

=cut


Makefile.PL
# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v6.025.
use strict;
use warnings;



use ExtUtils::MakeMaker;

my %WriteMakefileArgs = (
  "ABSTRACT" => "Work package for holiday modules in the Date::Holidays namespace",
  "AUTHOR" => "Jonas B. <jonasbn\@cpan.org>",
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0
  },
  "DISTNAME" => "Task-Date-Holidays",
  "LICENSE" => "artistic_2",
  "NAME" => "Task::Date::Holidays",
  "PREREQ_PM" => {
    "Date::Holidays::AT" => 0,
    "Date::Holidays::AU" => 0,
    "Date::Holidays::AW" => 0,
    "Date::Holidays::BR" => 0,
    "Date::Holidays::BY" => 0,
    "Date::Holidays::CA" => 0,
    "Date::Holidays::CA_ES" => 0,
    "Date::Holidays::CN" => 0,
    "Date::Holidays::CZ" => 0,
    "Date::Holidays::DE" => 0,
    "Date::Holidays::DK" => 0,
    "Date::Holidays::ES" => 0,
    "Date::Holidays::FR" => "0.04",
    "Date::Holidays::GB" => 0,
    "Date::Holidays::KR" => 0,
    "Date::Holidays::KZ" => 0,
    "Date::Holidays::NL" => 0,
    "Date::Holidays::NO" => 0,
    "Date::Holidays::NZ" => 0,
    "Date::Holidays::PL" => 0,
    "Date::Holidays::PT" => "0.03",
    "Date::Holidays::RU" => 0,
    "Date::Holidays::SK" => 0,
    "Date::Holidays::UA" => 0,
    "Date::Holidays::UK" => 0,
    "Date::Holidays::US" => 0,
    "Date::Holidays::USFederal" => 0,
    "Date::Japanese::Holiday" => 0
  },
  "TEST_REQUIRES" => {
    "Date::Holidays" => 0,
    "File::Spec" => 0,
    "IO::Handle" => 0,
    "IPC::Open3" => 0,
    "Pod::Coverage::TrustPod" => 0,
    "Test::Class" => 0,
    "Test::Fatal" => 0,
    "Test::Kwalitee" => "1.21",
    "Test::More" => 0,
    "Test::Pod" => "1.41",
    "Test::Pod::Coverage" => "1.08"
  },
  "VERSION" => "0.12",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


my %FallbackPrereqs = (
  "Date::Holidays" => 0,
  "Date::Holidays::AT" => 0,
  "Date::Holidays::AU" => 0,
  "Date::Holidays::AW" => 0,
  "Date::Holidays::BR" => 0,
  "Date::Holidays::BY" => 0,
  "Date::Holidays::CA" => 0,
  "Date::Holidays::CA_ES" => 0,
  "Date::Holidays::CN" => 0,
  "Date::Holidays::CZ" => 0,
  "Date::Holidays::DE" => 0,
  "Date::Holidays::DK" => 0,
  "Date::Holidays::ES" => 0,
  "Date::Holidays::FR" => "0.04",
  "Date::Holidays::GB" => 0,
  "Date::Holidays::KR" => 0,
  "Date::Holidays::KZ" => 0,
  "Date::Holidays::NL" => 0,
  "Date::Holidays::NO" => 0,
  "Date::Holidays::NZ" => 0,
  "Date::Holidays::PL" => 0,
  "Date::Holidays::PT" => "0.03",
  "Date::Holidays::RU" => 0,
  "Date::Holidays::SK" => 0,
  "Date::Holidays::UA" => 0,
  "Date::Holidays::UK" => 0,
  "Date::Holidays::US" => 0,
  "Date::Holidays::USFederal" => 0,
  "Date::Japanese::Holiday" => 0,
  "File::Spec" => 0,
  "IO::Handle" => 0,
  "IPC::Open3" => 0,
  "Pod::Coverage::TrustPod" => 0,
  "Test::Class" => 0,
  "Test::Fatal" => 0,
  "Test::Kwalitee" => "1.21",
  "Test::More" => 0,
  "Test::Pod" => "1.41",
  "Test::Pod::Coverage" => "1.08"
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);
